deploy-board/deploy_board/webapp/group_view.py

```diff
@@ -1045,16 +1045,20 @@ def add_instance(request, group_name):
                 placement_group = params['placementGroup']
                 use_placement_group = True
 
-        # Check if asg is enabled and does not use placement group. Then launch in asg
-        if str(asg_status).upper() == "ENABLED" and not use_placement_group:
+        # When a group has an associated ASG, its status is either ENABLED or DISABLED.
+        # We should always launch in ASG when an ASG is set up regardless its status,
+        # unless placement group is specified.
+        if (asg_status == 'ENABLED' or asg_status == 'DISABLED') and not use_placement_group:
             launch_in_asg = True
 
         if launch_in_asg:
```

Comment:
looks like we don't need extra launch_in_asg variable. lines 1051-1054 can be simplified with single if condition.

Reply:
also, do we still need to check for asg_status given it's launching instance for either ENABLED or DISABLED ?

can be just "if not use_placement_group:" ?

Reply:
> can be just "if not use_placement_group:" ?
No, because if `asg_status == 'UNKNOWN'` and `use_placement_group == False`, the current condition won't be met, but condition`if not use_placement_group` would be metand the code would increase the capacity of a non-exist ASG.

Reply:
Removed `launch_in_asg`.